frontend/src/features/map_planning/components/timeline/ItemSliderPicker.tsx

```diff
@@ -0,0 +1,307 @@
+    }
+  };
+
+  const handleMouseMove = (e: MouseEvent) => {
+    if (!isDragging.current || dragStartX.current === null) return;
+
+    const container = sliderContainerRef.current;
+    if (container && scrollPositionOnMouseDown.current) {
+      const deltaX = e.clientX - dragStartX.current;
+
+      container.scrollTo({
+        behavior: 'auto',
+        left: scrollPositionOnMouseDown.current - deltaX,
+      });
+    }
+  };
+
+  const handleItemClick = (e: React.MouseEvent, index: number) => {
+    const container = sliderContainerRef.current;
+    if (container && scrollPositionOnMouseDown.current === container.scrollLeft) {
+      setSelectedItemAndNotifyParent(items[index].key);
+      scrollToIndex(index);
+    }
+  };
+
+  const handleMouseUp = () => {
+    if (
+      isDragging.current &&
+      scrollPositionOnMouseDown.current !== sliderContainerRef.current?.scrollLeft
+    ) {
+      scrollToIndex(getSelectedItemIndex());
+      onChange(selectedItemKey);
+    }
+
+    isDragging.current = false;
+    dragStartX.current = null;
+  };
+
+  const handleMouseWheel = (e: WheelEvent) => {
+    const container = sliderContainerRef.current;
+    if (container) {
+      e.preventDefault();
+
+      if (e.deltaY > 0) {
+        selectPreviousItem();
+      } else {
+        selectNextItem();
+      }
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'ArrowLeft') {
+      selectPreviousItem();
+      e.stopPropagation();
+      e.preventDefault();
+    } else if (e.key === 'ArrowRight') {
+      selectNextItem();
+      e.stopPropagation();
+      e.preventDefault();
+    }
+  };
+
+  useEffect(() => {
+    const container = sliderContainerRef.current;
+    container?.addEventListener('scroll', handleScroll);
+    container?.addEventListener('mousedown', handleMouseDown);
+    container?.addEventListener('wheel', handleMouseWheel);
+    document.addEventListener('mousemove', handleMouseMove);
+    document.addEventListener('mouseup', handleMouseUp);
+
+    return () => {
+      container?.removeEventListener('scroll', handleScroll);
+      container?.removeEventListener('mousedown', handleMouseDown);
+      container?.removeEventListener('wheel', handleMouseWheel);
+      document.removeEventListener('mousemove', handleMouseMove);
+      document.removeEventListener('mouseup', handleMouseUp);
+    };
+  });
+
+  // Add a useEffect hook to scroll to the selected item when items change from the outside.
+  useEffect(() => {
+    const selectedIndex = getSelectedItemIndex();
+    const selectedItemIndex = Math.min(Math.max(selectedIndex, 0), items.length - 1);
+    scrollToIndex(selectedItemIndex);
+  }, [items, getSelectedItemIndex]);
+
+  // Add a useEffect hook to update the selected item when the selected value changes from the outside.
+  useEffect(() => {
+    if (value === undefined) return;
+
+    setSelectedItemKey(value);
+    scrollToIndex(items.findIndex((item) => item.key === value));
```

Comment:
check how often scrollToIndex is called.
I think there might be some redundant calls when value is defined and the items change.

Reply:
you are right, there a several unnecessary calls. i created a left over issue to optimize the the calls

Reply:
You can also link from such conversations to the issue. I assume you are referring to #1107